lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -42,18 +42,29 @@ UsdUndoRenameCommand::UsdUndoRenameCommand(const UsdSceneItem::Ptr& srcItem, con
 	: Ufe::UndoableCommand()
 {
 	const UsdPrim& prim = srcItem->prim();
-	fStage = prim.GetStage();
-	fUfeSrcItem = srcItem;
-	fUsdSrcPath = prim.GetPath();
-	// Every call to rename() (through execute(), undo() or redo()) removes
+	m_fStage = prim.GetStage();
+	m_fUfeSrcItem = srcItem;
+	m_fUsdSrcPath = prim.GetPath();
+	
+    // Every call to rename() (through execute(), undo() or redo()) removes
 	// a prim, which becomes expired.  Since USD UFE scene items contain a
 	// prim, we must recreate them after every call to rename.
-	fUsdDstPath = prim.GetParent().GetPath().AppendChild(TfToken(newName.string()));
-	fLayer = defPrimSpecLayer(prim);
-	if (!fLayer) {
+	m_fUsdDstPath = prim.GetParent().GetPath().AppendChild(TfToken(newName.string()));
+	m_fLayer = defPrimSpecLayer(prim);
+	if (!m_fLayer) {
 		std::string err = TfStringPrintf("No prim found at %s", prim.GetPath().GetString().c_str());
 		throw std::runtime_error(err.c_str());
 	}
+
+    // check if the target layer has any opinions that affects selected prim
+    if (!isTargetLayerHaveOpinion(prim)) {
```

Comment:
So here if the edit target doesn't have an opinion, we tell the user to set the edit target to the strongest layer than does have an opinion.  O.K., maybe that makes sense, even if the strongest layer only has "over" opinion.

But what if the edit target is on a weak layer and the weak layer DOES have an opinion, that is overridden by an opinion in a stronger layer?  We're saying that's O.K.?  The user will rename in the weak layer, but the stronger layer will still have its opinion and it will be composed into the final result, no?  That doesn't seem to meet Will's acceptance criteria that "Trying to rename an object with opinions on stronger layers error out."  Perhaps what you're showing here is really work in progress and you're going to fix this?  Or is there something I haven't understood?

Reply:
Very valid question and I should have elaborated more on the change here:

if (!MayaUsdUtils::doesLayerHavePrimSpec(prim)) solves this request:

`Trying to rename an object that was not defined on the target layer errors out`

but as you also mentioned it doesn't really solve this request:

`Trying to rename an object with opinions on stronger layers error out.`

I am working on addressing this as we speak.

Reply:
@ppt-adsk please see 8f8295f3b8.